typescript/solver/solvers/BaseListener.ts

```diff
@@ -0,0 +1,68 @@
+import type { Provider } from "@ethersproject/providers";
+import { chainMetadata } from "@hyperlane-xyz/registry";
+import { MultiProvider } from "@hyperlane-xyz/sdk";
+import type { Contract, Signer } from "ethers";
+
+import type { Logger } from "../logger.js";
+import type { TypedEvent, TypedListener } from "../typechain/common.js";
+
+export abstract class BaseListener<
```

Comment:
Would still recommend calling this `OnChainBaseListener` since its assuming to get intents from on-chain (vs. an off-chain RFQ)